### ISSUE TITLE
Specify search parameters in `filter` field

### ISSUE THREAD

**Issue by 2e3s** [Rebilly/RebillyReportsAPI#10]
@RomanGotsiy 
We have a query parameter `filter` in format `filter=field:value;range:start..end;array=1,2,3`. There should be a way to describe all the filtering fields for some path.

**Comment by RomanGotsiy** [comment]
@2e3s 
I've checked and this format is impossible to document both in swagger 2.0 and swagger 3.0 and even in JsonSchema

So the only way I see is to document it just in parameter description. I understand that it will introduce duplications.
How critical is it? How many operations have this filter parameter. If there are just a few of them, I would recommend just to duplicate.
If there are many, then I can write another plugin for build system which will do all the dirty work.

**Comment by 2e3s** [comment]
Yes, my first thought was to write the list of acceptable filters right in the description. I hope swagger browser supports some formatting to output a formatted list of them. I think this will be enough for basic purpose, although having some syntax for it would allow some automated UI building.
As for how many endpoints have `filter`: literally, all resources have it when it comes to a list of entities. Some reports of this repository do as well.
There is also another query parameter `critera` which has a syntax much like https://github.com/Rebilly/RebillyAPI/tree/master/spec/definitions/Criteria but with a bit different possibilities.
No one just knows about it. 😸  I think it can hurt us because merchants will look in API when deciding if to use Rebilly, not see any filtering and move away because of that.

**Comment by RomanGotsiy** [comment]
>  I hope swagger browser supports some formatting to output a formatted list of them.
you can use markdown in description fields

> although having some syntax for it would allow some automated UI building.
For that you can just specify list of filterable fields in some vendor extension, e.g. `x-sortableFields`, or anything you can come up but starting with `x-`.

But for end-user I would recommend documenting this just in description using markdown.
